### DISCUSSION TITLE
Add macro？

### DISCUSSION THREAD

**Discussion by wujianming** [Macad3D/Macad3D#53] [General]
hi. Have you considered the implementation of macro, similar to freecad

**Comment by dipts** [comment]
Macros are - also with FreeCAD - nothing else than scripts. You can call in a Macad script all functions of the UI directly via commands, e.g. `Macad.Interaction.ModelCommands.CreateBox`.

There is currently no function for recording UI commands.

**Comment by wujianming** [comment]
The biggest benefit of macros is that they can record actions
